### PR TITLE
MULE-9290: Wilcard generics resolved to the right DataType

### PR DESCRIPTION
**Pull request by pablomusumeci**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java [hunk 1]
```diff
@@ -60,6 +62,8 @@
 public class IntrospectionUtils
 {
 
+    private static final Log logger = LogFactory.getLog(IntrospectionUtils.class);
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java]: always use slf4j loggers

File: modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java [hunk 2]
```diff
@@ -236,9 +223,17 @@ private static boolean isOperation(Class<?> rawClass)
         Class<?>[] types = new Class<?>[resolvableTypes.length];
         for (int i = 0; i < resolvableTypes.length; i++)
         {
-            types[i] = resolvableTypes[i].getRawClass();
+            if (resolvableTypes[i].getRawClass() == null)
+            {
+                logger.warn(String.format("WildCard used in extension definition: '%s'", resolvableTypes[i].toString()));
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java]: if (logger.isWarnEnabled())

File: tests/unit/src/main/java/org/mule/tck/testmodels/fruit/FruitBox.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class FruitBox
+{
+
+    private List<? extends Fruit> fruitLikeList = Collections.synchronizedList(new ArrayList<>());
```
Comment (marianogonzalez) [inline tests/unit/src/main/java/org/mule/tck/testmodels/fruit/FruitBox.java]: why do these lists need to be synchronized?

File: tests/unit/src/main/java/org/mule/tck/testmodels/fruit/FruitBox.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    private List<Fruit> fruitList = Collections.synchronizedList(new ArrayList<Fruit>());
+    private List<?> wildCardList = Collections.synchronizedList(new ArrayList<>());
+    private List rawList = Collections.synchronizedList(new ArrayList<>());
+
+    private Map<?, ?> wildCardMap = Collections.synchronizedMap(new HashMap<>());
```
Comment (marianogonzalez) [inline tests/unit/src/main/java/org/mule/tck/testmodels/fruit/FruitBox.java]: same as lists... why the synchronized?

File: modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java [hunk 5]
```diff
@@ -145,36 +149,19 @@ private static DataType toDataType(ResolvableType type)
...
 
         if (isOperation(rawClass))
         {
-            resolvedDataType = DataType.of(OperationModel.class);
+            return DataType.of(OperationModel.class);
         }
-        else if (List.class.isAssignableFrom(rawClass))
+        else if (ArrayUtils.isEmpty(generics))
         {
-            if (!ArrayUtils.isEmpty(generics) && generics[0].getRawClass() != null)
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java]: where did this behaviour go?

Comment (pablomusumeci) [inline modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java]: That behaviour is contained in this line -> resolvableTypes[i].resolve(Object.class);

which is called only if !ArrayUtils.isEmpty(generics)

**Pull request merged**